lib/app/person.ex

```diff
@@ -1,45 +1,74 @@
 defmodule App.Person do
   use Ecto.Schema
-  import Ecto.Changeset
+  import Ecto.{Changeset, Query}
   alias App.Repo
-  # https://stackoverflow.com/a/47501059/1148249
   alias __MODULE__
 
   schema "people" do
-    field :status, :id
-    field :tag, :id
+    field :auth_provider, :string
+    field :givenName, Fields.Encrypted
+    field :key_id, :integer
+    field :locale, :string
+    field :picture, Fields.Encrypted
+    field :status, :integer
+
     timestamps()
   end
 
-  @doc """
-  Default attributes validation for Person
-  """
+  def get_person!(id), do: Repo.get!(__MODULE__, id)
+
+  @doc false
   def changeset(person, attrs) do
     person
-    |> cast(attrs, [:id, :status, :tag])
+    |> cast(set_id(attrs), [
+      :givenName,
+      :auth_provider,
+      :picture,
+      :locale,
+      :status,
+      :id
+    ])
+    |> validate_required([:givenName, :auth_provider])
   end
 
-  def create_person(attrs) do
-    changeset(%Person{}, attrs)
-    |> Repo.insert!()
+  # workaround for people_pkey (unique_constraint)
+  # see: github.com/dwyl/app-mvp/issues/115
+  defp set_id(attrs) do
+    if Map.has_key?(attrs, :id) do
+      attrs
+    else
+      # https://til.hashrocket.com/posts/e0754031e3-counting-records-with-ecto
+      id = Repo.aggregate(from(p in "people"), :count, :id)
+      Map.put(attrs, :id, id)
+    end
   end
```

Comment:
Do we need this function? Because the id is coming from `auth` app and always exists after login, I think attrs.id should already be defined, no?

Reply:
Good question. Hopefully we won't need this entire _file_ ... #118 😉